src/common/HTTPRequest.ts

```diff
@@ -66,44 +143,120 @@ export class HTTPRequest {
       this._headers[key.toLowerCase()] = event.request.headers[key];
   }
 
+  /**
+   * @returns the URL of the request
+   */
   url(): string {
     return this._url;
   }
 
+  /**
+   * Contains the request's resource type as it was perceived by the rendering
+   * engine.
+   * @remarks
+   * @returns one of the following: `document`, `stylesheet`, `image`, `media`,
+   * `font`, `script`, `texttrack`, `xhr`, `fetch`, `eventsource`, `websocket`,
+   * `manifest`, `other`.
+   */
   resourceType(): string {
+    // TODO (@jackfranklin): protocol.d.ts has a type for this, but all the
+    // string values are uppercase. The Puppeteer docs explicitly say the
+    // potential values are all lower case, and the constructor takes the event
+    // type and calls toLowerCase() on it, so we can't reuse the type from the
+    // protocol.d.ts. Why do we lower case?
```

Comment:
@mathiasbynens I don't suppose you have any insight here?

Reply:
I suppose for parity with other Puppeteer APIs? We don't really use uppercase string values anywhere.